admin/preprints/views.py

```diff
@@ -102,14 +102,16 @@ def update_subjects_for_provider(self, request, old_provider, new_provider):
             else:
                 new_subject = new_provider.subjects.filter(bepress_subject_id=bepress_id)
             if not new_subject.exists():
-                subject_problems = subject_problems + 1
+                subject_problems.append(subject.text)
                 new_subject = subject
             else:
                 new_subject = new_subject[0]
             new_subjects.append(new_subject.hierarchy)
         self.object.set_subjects(new_subjects, Auth(request.user))
         if subject_problems:
-            messages.warning(request, 'Unable to find subjects in new provider for {} subject(s)'.format(subject_problems))
+            messages.warning(request, 'Unable to find subjects in new provider for the following subject(s):')
```

Comment:
```suggestion
            messages.warning(request, 'Unable to find subjects in new provider for the following subject(s): ' + ', '.join(subject_problems))
```

I think this looks a bit better on the Admin App. If making this change, the for loop on lines 113-114 can be deleted.

Reply:
Corbin made the good point that the way the subjects are currently presented (as separate bullet points) can make them easier to read, especially if there are a lot of subjects that couldn't be updated so I don't think abiding by this suggested change is necessary.